semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/postgres-connector.md

```diff
@@ -130,66 +126,422 @@ It is possible to construct a direct reference to a named collection with a cust
+    {
+        // Split the token into its parts (Header, Payload, Signature)
+        var tokenParts = jwtToken.Split('.');
+        if (tokenParts.Length != 3)
+        {
+            return null;
+        }
+
+        // The payload is the second part, Base64Url encoded
+        var payload = tokenParts[1];
+
+        // Add padding if necessary
+        payload = AddBase64Padding(payload);
+
+        // Decode the payload from Base64Url
+        var decodedBytes = Convert.FromBase64String(payload);
+        var decodedPayload = Encoding.UTF8.GetString(decodedBytes);
+
+        // Parse the decoded payload as JSON
+        var payloadJson = JsonSerializer.Deserialize<JsonElement>(decodedPayload);
+
+        // Try to get the username from 'upn', 'preferred_username', or 'unique_name' claims
+        if (payloadJson.TryGetProperty("upn", out var upn))
+        {
+            return upn.GetString();
+        }
+        else if (payloadJson.TryGetProperty("preferred_username", out var preferredUsername))
+        {
+            return preferredUsername.GetString();
+        }
+        else if (payloadJson.TryGetProperty("unique_name", out var uniqueName))
+        {
+            return uniqueName.GetString();
+        }
+
+        return null;
+    }
+
+    private static string AddBase64Padding(string base64) => (base64.Length % 4) switch
+    {
+        2 => base64 + "==",
+        3 => base64 + "=",
+        _ => base64,
+    };
+}
+```
+
+Now you can use the `UseEntraAuthentication` method to set up the connection string for the Postgres connector:
+
+```csharp
+using Microsoft.SemanticKernel.Connectors.Postgres;
+
+var connectionString = "Host=mydb.postgres.database.azure.com;Port=5432;Database=postgres;SSL Mode=Require;";  // No Username or Password
+var dataSourceBuilder = new NpgsqlDataSourceBuilder(connectionString);
+dataSourceBuilder.UseEntraAuthentication();
+dataSourceBuilder.UseVector();
+using var dataSource = dataSourceBuilder.Build();
+
+var vectorStore = new PostgresVectorStore(dataSource);
+```
+
+By default, the `UseEntraAuthentication` method uses the [DefaultAzureCredential](/dotnet/api/azure.identity.defaultazurecredential) to authenticate with Azure AD. You can also provide a custom `TokenCredential` implementation if needed.
+
 ::: zone-end
 ::: zone pivot="programming-language-python"
 
-## Coming soon
+## Getting started
+
+Install semantic kernel with the Postgres extras, which includes the [Postgres client](https://github.com/Postgres/Postgres-client).
+
+```cli
+pip install semantic-kernel[postgres]
+```
+
+You can then create a vector store instance using the `PostgresStore` class.
+You can pass a `psycopg_pool` [AsyncConnectionPool](https://www.psycopg.org/psycopg3/docs/api/pool.html#psycopg_pool.AsyncConnectionPool) directly,
+or use the `PostgresSettings` to create a connection pool from environment variables.
+
+```python
+from semantic_kernel.connectors.memory.postgres import PostgresStore, PostgresSettings
```

Comment:
We moved a bunch of things around so please have a look, every import should now come from semantic_kernel.connectors.postgres

Reply:
Fixed in  755d00f